Essentials/src/com/earth2me/essentials/EssentialsBlockListener.java

```
@@ -27,41 +27,39 @@ public EssentialsBlockListener(final IEssentials ess)
 	@EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
 	public void onBlockPlace(final BlockPlaceEvent event)
 	{
-		// Do not rely on getItemInHand();
-		// http://leaky.bukkit.org/issues/663
-		final ItemStack is = LocationUtil.convertBlockToItem(event.getBlockPlaced());
+		final ItemStack is = event.getPlayer().getItemInHand();
 		if (is == null)
 		{
 			return;
 		}
 
-		if (is.getType() == Material.MOB_SPAWNER && event.getItemInHand() != null && event.getPlayer() != null
-			&& event.getItemInHand().getType() == Material.MOB_SPAWNER)
+		if (is.getType() == Material.MOB_SPAWNER)
 		{
 			final BlockState blockState = event.getBlockPlaced().getState();
-			if (blockState instanceof CreatureSpawner)
+			final CreatureSpawner spawner = (CreatureSpawner)blockState;
+			final EntityType type = EntityType.fromId(is.getDurability());
```

getDurability() while not marked as deprecated is no better than the method you replaced here, and will cause silent failures if the underlying code is changed, this is a bad swap.